lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -662,51 +669,32 @@ def unblock_command(cmd)
   # This code was originally taken from https://github.com/ruby/ruby/blob/93420d34aaf8c30f11a66dd08eb186da922c831d/lib/shellwords.rb#L88
   #
   def shellsplitex(line)
-    quote = nil
-    words = []
+    unbalanced_quote = nil
+    tokens = []
     field = String.new
-    regexp = %r{
-      \G\s*(?>(?<word>[^\s\'\"]+)             # Words within str
```

Comment:
These comments were added by https://github.com/rapid7/metasploit-framework/pull/15050, but the changes broke the regex. I've reverted it to the original regex, but now I'm additionally matching preceding whitespace to correctly track the token index